src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -94,7 +94,20 @@ public async Task<GeneratedCode> GenerateAsync(CancellationToken cancellationTok
                 var newDocument = callSiteDocument.Document.WithSyntaxRoot(newCallSiteRoot.ReplaceNode(destination, newContainer));
                 newDocument = await Simplifier.ReduceAsync(newDocument, Simplifier.Annotation, null, cancellationToken).ConfigureAwait(false);
 
-                var finalDocument = await SemanticDocument.CreateAsync(newDocument, cancellationToken).ConfigureAwait(false);
+                var generatedDocument = await SemanticDocument.CreateAsync(newDocument, cancellationToken).ConfigureAwait(false);
+
+                // For nullable reference types, we can provide a better experience by reducing use 
+                // of nullable reference types after a method is done being generated. If we can
+                // determine that the method never returns null, for example, then we can
+                // make the signature into a non-null reference type even though
+                // the original type was nullable. This allows our code generation to
+                // follow our recommendation of only using nullable when necessary.
+                // This is done after method generation instead of at analyzer time because it's purely
+                // based on the resulting code, which the generator can modify as needed. If return statements
+                // are added, the flow analysis could change to indicate something different. It's cleaner to rely
```

Comment:
I'm still not sure I understand the case where that would happen. That would seem like the transformation we're doing in extract method has some deeper issue.